### DISCUSSION TITLE
How do i manually change my network interfaces

### DISCUSSION THREAD

**Discussion by iqworks** [Security-Onion-Solutions/securityonion#5592] [Unsupported Versions] [answered]
I am using VMware workstation 16 pro
 I am using Wazuh 2.3.70 with centOS 7.
I am trying to see the network interfaces.

 After searching for "How do i manually change my network interfaces ", the only thing i found was this :  " https://github.com/Security-Onion-Solutions/securityonion/discussions/4286 ". it says :
  "I have the very latest version that I cloned from Github today. I would like to go back and change the settings for my network interfaces. Where are the files that contain this info? Or is there a script to run? so-setup-network will take me through the whole installation."
   This was the only answer I saw:
“not supported yet”? 
  If this is not the case anymore, may I ask if it is supported by now?
  
  My primary confusion is that i see two command structures that lead me to believe that at least one of them is the way to edit my security onion /etc/network/interfaces file.  
   
  The first command structure I see is used in debian. I just need to know If I can use this to edit my security onion interfaces file (https://wiki.debian.org/NetworkConfiguration)?

   The second command structure looks like this : 
"/etc/sysconfig/network-scripts/" . 
   I see this when i use this command structure:
 With this command “cat /etc/sysconfig/network-scripts/ifcfg-* ” I see :
![image](https://user-images.githubusercontent.com/5569263/134087914-84d5438d-1458-441d-be47-83a51d0bb91f.png)
![image](https://user-images.githubusercontent.com/5569263/134087935-6c445eeb-cb74-4b91-8369-3942b4379da2.png)
![image](https://user-images.githubusercontent.com/5569263/134087951-413f1193-14a3-4041-88c5-6453c50e4b26.png)
![image](https://user-images.githubusercontent.com/5569263/134087959-93af8cf1-0649-4b03-80e8-0bce1cf44803.png)
![image](https://user-images.githubusercontent.com/5569263/134087980-1ab6b572-fc31-4a41-b37f-ff4d6941318f.png)

  However, when i look in here, 
https://docs.securityonion.net/_/downloads/en/2.3/pdf/ 
  I only see one reference to "/etc/nework/interfaces"
  
![image](https://user-images.githubusercontent.com/5569263/134088286-44991c80-dd49-4ad7-814a-f60e629bcc56.png)

  So, in trying to keep within the vein of this post, I am wondering:
1 - how to access my SO V2.3.70 interfaces file. 
2 - what is the minumum NIC interfaces I need to run security onion properly.  
3 - should my SO managment IP always be my NAT (192.168.80.0 in VMware)?
4 - should my allow (sniffer) IP be one of the VMnets not used by anything else?
5 - Is there a GUI network manager for my version of security onion?
6 - I also read that my security onion works best with a wired connection? But my host is windows 10 using wi-fi. I think i have this confused, does this mean I need to use a wired ehternet for my VMs or is my current NAT and bridge switches ok to use to access the internet to allow security onion to work properly. Sounds like a silly question, but I just want to make sure that since I am new to this stuff that I have it right, when it comes to security onion?

 thanks for any help or advice

**Comment by dougburks** [comment, marked as answer]
> 1 - how to access my SO V2.3.70 interfaces file.

If you're using our ISO image which is based on CentOS 7 then `/etc/sysconfig/network-scripts/` is the correct place to look for where CentOS defines network interfaces. However, please keep in mind that the IP address that you set during installation is also referenced in other places in the Security Onion configuration. So if you need to change your IP address after installation (which we generally recommend against), you would need to change it at the OS level and then need to run the experimental `so-ip-update` utility to update all of those other references:
https://docs.securityonion.net/en/2.3/ip.html

> 2 - what is the minumum NIC interfaces I need to run security onion properly.

Depends on configuration. From https://docs.securityonion.net/en/2.3/hardware.html#nic:
```
You’ll need at least one wired network interface dedicated to management (preferably connected to a dedicated management network). We recommend using static IP addresses where possible.

If you plan to sniff network traffic from a tap or span port, then you will need one or more interfaces dedicated to sniffing (no IP address).
```

> 3 - should my SO managment IP always be my NAT (192.168.80.0 in VMware)?

Your management IP address should be a valid IP address in whatever network range it is connected to.  If your management NIC is connected to a VMware NAT network, then it should be a valid IP address in that NAT network range.

> 4 - should my allow (sniffer) IP be one of the VMnets not used by anything else?

I'm not sure I understand the question. Are you asking about the sniffing interface? From https://docs.securityonion.net/en/2.3/hardware.html#nic:
```
If you plan to sniff network traffic from a tap or span port, then you will need one or more interfaces dedicated to sniffing (no IP address).
```

> 5 - Is there a GUI network manager for my version of security onion?

If using our ISO image, you should set your IP address once in the installer and should never have to touch it again.

> 6 - I also read that my security onion works best with a wired connection? But my host is windows 10 using wi-fi. I think i have this confused, does this mean I need to use a wired ehternet for my VMs or is my current NAT and bridge switches ok to use to access the internet to allow security onion to work properly. Sounds like a silly question, but I just want to make sure that since I am new to this stuff that I have it right, when it comes to security onion?

If you're running in VMware and have configured the VM with standard VMware network interfaces, then those network interfaces are seen as wired interfaces by your Security Onion VM.

**Comment by iqworks** [comment]
Doug, thank you SO much for these answers, cleared up a lot.
